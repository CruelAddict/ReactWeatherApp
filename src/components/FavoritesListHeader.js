import React from 'react'

export default (props) => {
    return <div className={"favorites-header-container"}>
        <div className={"favorites-title-container"}>
            <h2>Избранное</h2>
        </div>
        <form className={"new-favorite-form"} onSubmit={props.handleSubmit}>
            <input
                placeholder={"Добавить новый город"}
                className={"add-favorite-input"}
                value={props.state.favoritesInput}
                onChange={props.handleChange}
            />
            <button className={"favorites-action-button"}>+</button>
        </form>
    </div>

}
