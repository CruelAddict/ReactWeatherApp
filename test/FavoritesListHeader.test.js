import React from 'react';
import { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import * as Enzyme from "enzyme";
Enzyme.configure({ adapter: new Adapter() });

import FavoritesListHeader from '../src/components/FavoritesListHeader';


describe ('FavoritesListHeader', () => {
    it('Should render successfully', () => {
        const header = shallow(
            <FavoritesListHeader state={{favoritesInput: ""}}/>
        );
        expect(header).toMatchSnapshot()
    });

});
