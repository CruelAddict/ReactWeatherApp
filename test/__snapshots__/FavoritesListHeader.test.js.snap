// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesListHeader Should render successfully 1`] = `
<div
  className="favorites-header-container"
>
  <div
    className="favorites-title-container"
  >
    <h2>
      Избранное
    </h2>
  </div>
  <form
    className="new-favorite-form"
  >
    <input
      className="add-favorite-input"
      placeholder="Добавить новый город"
      value=""
    />
    <button
      className="favorites-action-button"
    >
      +
    </button>
  </form>
</div>
`;
